### ISSUE TITLE
Support mac system?

### ISSUE THREAD

**Issue by mozillazg** [thombashi/pingparsing#28]
First of all, I would like to thank you for your work!  👍 

Do you have any plans for support mac system?

**Comment by thombashi** [comment]
Yes.
I planning to support for MAC.
However, I could not promise exactly when.
Since I have not possessed any MAC environment for now.

If you could provide me some of the `ping` command results (success cases/error cases/etc.) at MAC, 
I can support MAC system more than faster.

**Comment by mozillazg** [comment]
arguments:
```
$ ping -h
ping: option requires an argument -- h
usage: ping [-AaDdfnoQqRrv] [-c count] [-G sweepmaxsize]
            [-g sweepminsize] [-h sweepincrsize] [-i wait]
            [-l preload] [-M mask | time] [-m ttl] [-p pattern]
            [-S src_addr] [-s packetsize] [-t timeout][-W waittime]
            [-z tos] host
       ping [-AaDdfLnoQqRrv] [-c count] [-I iface] [-i wait]
            [-l preload] [-M mask | time] [-m ttl] [-p pattern] [-S src_addr]
            [-s packetsize] [-T ttl] [-t timeout] [-W waittime]
            [-z tos] mcast-group
Apple specific options (to be specified before mcast-group or host like all options)
            -b boundif           # bind the socket to the interface
            -k traffic_class     # set traffic class socket option
            -K net_service_type  # set traffic class socket options
            -apple-connect       # call connect(2) in the socket
            -apple-time          # display current time
```

error:

```
$ ping twitter.com -q -w 10
ping: invalid option -- w
```

```
$ ping twitter.com -q -W 10            # need cancel by control c
PING twitter.com (59.24.3.173): 56 data bytes
^C
--- twitter.com ping statistics ---
59 packets transmitted, 0 packets received, 100.0% packet loss
```

```
$ ping twitter.com -q -W 10 -c 10
PING twitter.com (31.13.78.66): 56 data bytes

--- twitter.com ping statistics ---
10 packets transmitted, 0 packets received, 100.0% packet loss
```

```
$ ping twitter.com -q -t 10
PING twitter.com (69.171.234.18): 56 data bytes

--- twitter.com ping statistics ---
10 packets transmitted, 0 packets received, 100.0% packet loss
```

success:

```
$ ping -q -t 10 github.com
PING github.com (192.30.255.112): 56 data bytes

--- github.com ping statistics ---
9 packets transmitted, 8 packets received, 11.1% packet loss
round-trip min/avg/max/stddev = 198.739/449.140/2012.432/594.003 ms
```

```
$ ping github.com -q -t 10
PING github.com (192.30.255.113): 56 data bytes

--- github.com ping statistics ---
10 packets transmitted, 10 packets received, 0.0% packet loss
round-trip min/avg/max/stddev = 218.391/283.477/405.879/70.170 ms
```

**Comment by mozillazg** [comment]
on some mac system `ping {hostname} -q -t` will raise an error because can't put arguments after hostname:

```
$ ping github.com -q -t 10
usage: ping [-AaDdfnoQqRrv] [-b boundif] [-c count] [-G sweepmaxsize]
            [-g sweepminsize] [-h sweepincrsize] [-i wait] [−k trafficclass]
            [-l preload] [-M mask | time] [-m ttl] [-p pattern]
            [-S src_addr] [-s packetsize] [-t timeout][-W waittime] [-z tos]
            host
       ping [-AaDdfLnoQqRrv] [-b boundif] [-c count] [-I iface] [-i wait]
            [−k trafficclass] [-l preload] [-M mask | time] [-m ttl] [-p pattern] [-S src_addr]
            [-s packetsize] [-T ttl] [-t timeout] [-W waittime]
            [-z tos] mcast-group
$ echo $?
64
```

**Comment by thombashi** [comment]
@mozillazg 
Thank for your information.
That would be a great help.
I will try to implement support for MAC by your information.
(Probably in this weekend)

**Comment by thombashi** [comment]
@mozillazg 
I had implemented MAC support at `pingparsing 0.11.0`
Please let me know if you have any problems at MAC environments.

**Comment by mozillazg** [comment]
@thombashi Thanks for your work. It worked as expected. 👍
